rerun_py/src/catalog/connection_handle.rs

```diff
@@ -401,62 +405,188 @@ impl ConnectionHandle {
-                })?;
+                .map_err(Into::<re_grpc_client::StreamError>::into)
+                .map_err(to_py_err)??;
+
+                // Then we need to fully decode these chunks, i.e. both the transport layer (Protobuf)
+                // and the app layer (Arrow).
+                let mut chunk_stream =
+                    re_grpc_client::get_chunks_response_to_chunk_and_partition_id(
+                        get_chunks_response_stream,
+                    );
+
+                let (tx, mut rx) = tokio::sync::mpsc::channel(32); // 32 batches of chunks, not 32 chunks
+
+                // We want the underlying HTTP2 client to keep polling on the gRPC stream as fast
+                // as non-blockingly possible, which cannot happen if we just poll once in a while
+                // in-between decoding phases. This results in the stream just sleeping, waiting
+                // for IO to complete, way more frequently that it should.
+                // We resolve that by spawning a dedicated I/O task that just polls the stream as fast as
+                // the stream will allows. This way, whenever the underlying HTTP2 stream is polled, we
+                // will already have pre-fetched a bunch of data for it.
+                tokio::spawn(
+                    async move {
+                        while let Some(chunk_and_partition_id) = chunk_stream.next().await {
+                            // The only possible error is the other end hanging up, which is not our problem.
+                            if tx.send(chunk_and_partition_id).await.is_err() {
+                                break;
+                            }
+                        }
+                    }
+                    .instrument(tracing::trace_span!("get_chunks::forward"))
+                    .in_current_span(),
+                );
 
-                let store = stores.entry(partition_id.clone()).or_insert_with(|| {
-                    let store_info = StoreInfo {
-                        application_id: ApplicationId::from(partition_id),
-                        store_id: StoreId::random(StoreKind::Recording),
-                        cloned_from: None,
-                        store_source: StoreSource::Unknown,
-                        store_version: None,
-                    };
+                let mut stores = BTreeMap::default();
+
+                while let Some(chunks_and_partition_ids) = rx.recv().await {
+                    // We want to make sure to offload that compute-heavy work to the compute worker pool: it's
+                    // not going to make this one single pipeline any faster, but it will prevent starvation of
+                    // the Tokio runtime (which would slow down every other futures currently scheduled!).
+                    stores = tokio::task::spawn_blocking({
+                        let mut stores = stores.clone(); // just cloning handles
```

Comment:
I missed the leading `stores =` on first read and thought I was losing my mind for a bit.

```suggestion
                    stores = tokio::task::spawn_blocking({
                        // Clone the stores for mutabillity within the spawned task.
                        // Note at the end of this task we return the mutated stores and assign
                        // it back to the outer stores variable.
                        let mut stores = stores.clone(); 
```